File: 프로그래머스연습문제/문자열 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool compare(char a, char b){
    
    if(a >= 'A' && a <='Z' && b >= 'A' && b <='Z') return a > b;
    
    else if(a >= 'A' && a <='Z' && b >= 'a' && b <='z') return false;
    
    else if(a >= 'a' && a <='z' && b >= 'A' && b <='Z') return true;
    
    else{
        return a > b;
    }
}

string solution(string s) {
    string answer = "";
    sort(s.begin(),s.end(),compare);
    answer = s;
    
    
    return answer;
}

File: 프로그래머스_python/연습문제/Level1/문자열 내림차순으로 배치하기.py
def solution(s):
    answer = ''
    
    upper_array = []
    lower_array = []
    
    for i in range(len(s)):
        if s[i].isupper() == True:
            upper_array.append(s[i])
        else:
            lower_array.append(s[i])
    
    upper_array.sort(reverse = True)
    lower_array.sort(reverse = True)
    
    upper_array = ''.join(upper_array)
    lower_array = ''.join(lower_array)
    
    answer = lower_array + upper_array
    
    
    return answer
